### PR TITLE
Fix: uri parser authSource and dbName bug

### PR DESCRIPTION
**Pull request by erfanium**
Currently when authSource and dbName comes together, parse ignores `dbName` that is not expected

### CODE REVIEW

File: src/utils/uri.ts [hunk 1]
```diff
@@ -122,18 +122,22 @@ export type SrvConnectOptions = Omit<ConnectOptions, "servers"> & {
 
 export function parseSrvUrl(url: string): SrvConnectOptions {
   const data = parse_url(url);
+
+  const path = (data.pathname && (data.pathname.length > 1))
```
Comment (manyuanrong) [inline src/utils/uri.ts:126]: I think the variable name `path` is not very appropriate

Comment (erfanium) [reply src/utils/uri.ts:126]: What's your suggestion?
path of `mongodb://a:b@localhost:27017/dbName?authSource=admin2` uri is `/dbName`. That makes sense to 
Screenshot from 2021-09-22 18-20-41
me

Comment (manuelaguirre) [reply src/utils/uri.ts:126]: Maybe pathExists if I understood correctly that it's a boolean.

Comment (erfanium) [reply src/utils/uri.ts:126]: It's not boolean, it's string

Comment (manuelaguirre) [reply src/utils/uri.ts:126]: Sorry github just cut the code sample after line 126

Comment (manyuanrong) [reply src/utils/uri.ts:126]: Although this part of the uri is called `path`, it is in the connection string protocol of mongodb. Called `defaultAuthDB`
https://docs.mongodb.com/manual/reference/connection-string/#std-label-connections-standard-connection-string-format

**Comment by erfanium** [comment]
I also changed teh default db name from `admin` to `test`. That's the behavior from official mongodb driver for node

**Review by nerdkid93 — APPROVED** [review]
I just encountered this error in the `mongo@v0.27.0` release, and when I point my import map to your branch instead of `deno.land`, the auth issues I was encountering are fixed (although it might be worth updating the `README.md` Example to demonstrate the new behavior)

**Comment by erfanium** [comment]
@nerdkid93 Yes, we should notify this change with release note.

**Review by manyuanrong — APPROVED** [review]

**Pull request merged**